### PR TITLE
Ticker 4: Create all models and fake data

### PR DESCRIPTION
**Pull request by BaoVu0712**
### Description
- Create all models and fake data

### Include
- Within Domain package, create models
- Within Test package, create fake data ( mock into UI )
- replace `ui` package into `feature`
- Analys form management Warehouse

### CODE REVIEW

File: app/src/main/java/com/example/warehousemanagement/domain/model/Address.kt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package com.example.warehousemanagement.domain.model
+
+data class Address(
```
Comment (PhunnMoc) [inline app/src/main/java/com/example/warehousemanagement/domain/model/Address.kt:3]: `id` field is verry necessary

File: app/src/main/java/com/example/warehousemanagement/domain/model/Customer.kt [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package com.example.warehousemanagement.domain.model
+
+data class Customer(
+    val id: String,
+    val customerName: String,
+    val address: Address,
+
```
Comment (PhunnMoc) [inline app/src/main/java/com/example/warehousemanagement/domain/model/Customer.kt:7]: Remove this redundant line ( and reformat, in mac is `command` + `option`+ `l`. Please research in Win

File: app/src/main/java/com/example/warehousemanagement/domain/model/Genre.kt [hunk 3]
```diff
@@ -0,0 +1,6 @@
+package com.example.warehousemanagement.domain.model
+
+data class Genre(
+    val idGenre: String,
+    val genreName: String // Tên loại sản phẩm
```
Comment (PhunnMoc) [inline app/src/main/java/com/example/warehousemanagement/domain/model/Genre.kt:5]: Ủa genre có image ko ta

File: app/src/main/java/com/example/warehousemanagement/domain/model/Notification.kt [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package com.example.warehousemanagement.domain.model
+
+import java.util.Date
+
+data class Notification(
+    val id: String,
+    val title: String,
+    val senderId: String,
+    val receiverId: String,
```
Comment (PhunnMoc) [inline app/src/main/java/com/example/warehousemanagement/domain/model/Notification.kt:9]: Ủa gửi được cho đơn cá nhân hả

File: app/src/main/java/com/example/warehousemanagement/domain/model/Product.kt [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+    val productName: String, // Tên sản phẩm
+    val genre: Genre, // Loại sản phẩm
+    val quantity: Int, // Số lượng sản phẩm
+    val description: String, // Mô tả sản phẩm
+    val supplier: Supplier, // Thông tin nhà cung cấp
+    val isInStock: Boolean, // Tình trạng hàng hóa
+    val barcode: String, // Mã vạch
+    val area: Area, // Khu vực
+    val lastUpdated: Date // Ngày cập nhật cuối
+)
```
Comment (PhunnMoc) [inline app/src/main/java/com/example/warehousemanagement/domain/model/Product.kt:16]: Where is `image` field

File: app/src/main/java/com/example/warehousemanagement/domain/model/Report.kt [hunk 6]
```diff
@@ -0,0 +1,12 @@
+package com.example.warehousemanagement.domain.model
+
+import java.util.Date
+
+data class Report(
```
Comment (PhunnMoc) [inline app/src/main/java/com/example/warehousemanagement/domain/model/Report.kt:5]: Who send this `report`. Add id reporter

File: app/src/main/java/com/example/warehousemanagement/domain/model/Role.kt [hunk 7]
```diff
@@ -0,0 +1,6 @@
+package com.example.warehousemanagement.domain.model
+
+data class Role(
```
Comment (PhunnMoc) [inline app/src/main/java/com/example/warehousemanagement/domain/model/Role.kt:3]: Just 2 role, i thinks it should be Emun with parameter `roleName`

File: app/src/main/java/com/example/warehousemanagement/domain/model/Status.kt [hunk 8]
```diff
@@ -0,0 +1,6 @@
+package com.example.warehousemanagement.domain.model
+
+enum class Status(val value: String) {
```
Comment (PhunnMoc) [inline app/src/main/java/com/example/warehousemanagement/domain/model/Status.kt:3]: Why `NotificationType` do not have parameter value like this. ( Chọn 1 trong 2 cho đồng bộ)

File: app/src/main/java/com/example/warehousemanagement/domain/model/User.kt [hunk 9]
```diff
@@ -0,0 +1,10 @@
+package com.example.warehousemanagement.domain.model
+
+data class User(
```
Comment (PhunnMoc) [inline app/src/main/java/com/example/warehousemanagement/domain/model/User.kt:3]: Why we dont remove all field like `role`, `picture` into `information` data class nhỉ

File: app/src/main/java/com/example/warehousemanagement/domain/model/ImportPackages.kt [hunk 10]
```diff
@@ -0,0 +1,13 @@
...
+import java.util.Date
+
+data class ImportPackages(
+    val id: String,                // Định danh gói nhập
+    val packageName: String,        // Tên gói nhập
+    val importDate: Date,           // Ngày nhập hàng
+    val supplier: Supplier,         // Thông tin nhà cung cấp
+    val idReceivedBy: String,       // Người nhận hàng
+    val status: Status= Status.PENDING,  // Tình trạng gói nhập, default is PENDING
+    val note: String? = null        // Ghi chú, optional
```
Comment (PhunnMoc) [inline app/src/main/java/com/example/warehousemanagement/domain/model/ImportPackages.kt:12]: What does this package contain? . Which product and quatity.

File: app/src/main/java/com/example/warehousemanagement/domain/model/ExportPackages.kt [hunk 11]
```diff
@@ -0,0 +1,14 @@
+package com.example.warehousemanagement.domain.model
+
+import java.util.Date
+
+data class ExportPackages(
```
Comment (PhunnMoc) [inline app/src/main/java/com/example/warehousemanagement/domain/model/ExportPackages.kt:5]: Same issue with ImportP, please modiy it nha

File: app/src/main/java/com/example/warehousemanagement/domain/model/ImportPackages.kt [hunk 12]
```diff
@@ -0,0 +1,13 @@
...
+)
```
Comment (PhunnMoc) [inline app/src/main/java/com/example/warehousemanagement/domain/model/ImportPackages.kt:13]: After checked by worker, if there're some deficit ( thâm hụt) , remember it nha

File: app/src/main/java/com/example/warehousemanagement/domain/model/Information.kt [hunk 13]
```diff
@@ -0,0 +1,7 @@
+package com.example.warehousemanagement.domain.model
+
+data class Information(
```
Comment (PhunnMoc) [inline app/src/main/java/com/example/warehousemanagement/domain/model/Information.kt:3]: every data class should have index (`id`)

**Review by PhunnMoc — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/warehousemanagement/domain/model/Genre.kt (hunk 3 above)
Comment (BaoVu0712) [reply app/src/main/java/com/example/warehousemanagement/domain/model/Genre.kt:5]: Theo tui thì ko cần hình ảnh

File: app/src/main/java/com/example/warehousemanagement/domain/model/Notification.kt (hunk 4 above)
Comment (BaoVu0712) [reply app/src/main/java/com/example/warehousemanagement/domain/model/Notification.kt:9]: Để tui chỉnh lại, mỗi lần thông báo là thông báo all

File: app/src/main/java/com/example/warehousemanagement/domain/model/ImportPackages.kt (hunk 12 above)
Comment (BaoVu0712) [reply app/src/main/java/com/example/warehousemanagement/domain/model/ImportPackages.kt:13]: when there's some deficit, this package will be canceled

Comment (PhunnMoc) [reply app/src/main/java/com/example/warehousemanagement/domain/model/ImportPackages.kt:13]: No, we use it to report with supplier to get return money or sth else. Those packages are alread delived to  our warehouse, and i thinks all package will have deficit.

File: app/src/main/java/com/example/warehousemanagement/domain/model/Notification.kt (hunk 4 above)
Comment (PhunnMoc) [reply app/src/main/java/com/example/warehousemanagement/domain/model/Notification.kt:9]: Yub, this way is better for dev =)))

File: app/src/main/java/com/example/warehousemanagement/domain/model/Genre.kt (hunk 3 above)
Comment (PhunnMoc) [reply app/src/main/java/com/example/warehousemanagement/domain/model/Genre.kt:5]: Ok

File: app/src/main/java/com/example/warehousemanagement/domain/model/Status.kt (hunk 8 above)
Comment (BaoVu0712) [reply app/src/main/java/com/example/warehousemanagement/domain/model/Status.kt:3]: cái chỗ này nè

**Comment by BaoVu0712** [comment]
Done

File: app/src/main/java/com/example/warehousemanagement/domain/model/Address.kt (hunk 1 above)
Comment (BaoVu0712) [reply app/src/main/java/com/example/warehousemanagement/domain/model/Address.kt:3]: Done

File: app/src/main/java/com/example/warehousemanagement/domain/model/Customer.kt (hunk 2 above)
Comment (BaoVu0712) [reply app/src/main/java/com/example/warehousemanagement/domain/model/Customer.kt]: Done

File: app/src/main/java/com/example/warehousemanagement/domain/model/User.kt (hunk 9 above)
Comment (BaoVu0712) [reply app/src/main/java/com/example/warehousemanagement/domain/model/User.kt:3]: Done

File: app/src/main/java/com/example/warehousemanagement/domain/model/ImportPackages.kt (hunk 10 above)
Comment (BaoVu0712) [reply app/src/main/java/com/example/warehousemanagement/domain/model/ImportPackages.kt:15]: Done

File: app/src/main/java/com/example/warehousemanagement/domain/model/ExportPackages.kt (hunk 11 above)
Comment (BaoVu0712) [reply app/src/main/java/com/example/warehousemanagement/domain/model/ExportPackages.kt:5]: Done

File: app/src/main/java/com/example/warehousemanagement/domain/model/StorageLocation.kt [hunk 14]
```diff
@@ -0,0 +1,7 @@
+package com.example.warehousemanagement.domain.model
+
+class StorageLocation (
+    val id:String,
+    val storageLocationName: String, // Tên khu vực
+    val storageLocationImage:String // URL hình ảnh khu vực
```
[the remark this answers is not in the document]
Comment (BaoVu0712) [reply app/src/main/java/com/example/warehousemanagement/domain/model/StorageLocation.kt]: Done

File: app/src/main/java/com/example/warehousemanagement/domain/model/Supplier.kt [hunk 15]
```diff
@@ -0,0 +1,9 @@
+package com.example.warehousemanagement.domain.model
+
+data class Supplier(
+    val idSupplier: String, // Định danh nhà cung cấp
+    val name: String, // Tên nhà cung cấp
+    val email:String,
+    val address: Address, // Địa chỉ nhà cung cấp
+    val stability:Float// Độ ổn định
```
[the remark this answers is not in the document]
Comment (BaoVu0712) [reply app/src/main/java/com/example/warehousemanagement/domain/model/Supplier.kt]: Done

File: app/src/main/java/com/example/warehousemanagement/domain/model/ImportPackages.kt (hunk 10 above)
Comment (PhunnMoc) [reply app/src/main/java/com/example/warehousemanagement/domain/model/ImportPackages.kt:13]: Done what :( , add ` val listProduct: List<Product>`

Comment (PhunnMoc) [reply app/src/main/java/com/example/warehousemanagement/domain/model/ImportPackages.kt:13]: do the same thing with `export`

File: app/src/main/java/com/example/warehousemanagement/domain/model/ExportPackages.kt (hunk 11 above)
Comment (PhunnMoc) [reply app/src/main/java/com/example/warehousemanagement/domain/model/ExportPackages.kt:5]: add List product

**Review by PhunnMoc — CHANGES_REQUESTED** [review]

**Review by PhunnMoc — APPROVED** [review]

**Pull request merged**